tests/Predis/Connection/Aggregate/SentinelReplicationTest.php

```diff
@@ -44,7 +45,8 @@ public function testParametersForSentinelConnectionShouldNotUseDatabaseAndPasswo
 
         $parameters = $replication->getSentinelConnection()->getParameters()->toArray();
 
-        $this->assertArraySubset(array('database' => null, 'password' => null), $parameters);
+        $this->assertArrayHasKey('database', $parameters);
+        $this->assertArrayHasKey('password', $parameters);
```

Comment:
This change comes with a different (and wrong) behaviour compared to the original code.

We must test that `database` and `password` exist in the resulting parameters array and that **they both have a `NULL` value**, not just that they merely exist as you did in this change. If `database` and `password` are set in connection parameters passed for **sentinel nodes**, then we must clear their values ([see here](https://github.com/predis/predis/blob/94ccb2cc3b8406a9bc54618cfbed37a224d7cb6c/src/Connection/Aggregate/SentinelReplication.php#L242-L245)). This will probably change later, at least for the `password` parameter, considering that that they added authentication for sentinels in Redis at last (see #594).

PS: just a personal thought, PHPUnit deprecated `assertArraySubset()` with 8.x and removed it with 9.x but I think it's an useful assertion and apparently [someone agrees with me](https://packagist.org/packages/dms/phpunit-arraysubset-asserts). Just saying, in our case it was used only in this test so we can easily forget about it.

Reply:
The package you referred to works with PHP 7.3+ not PHP 7.2.

Reply:
No there's no need to add that package, it was more of a personal side note on the decision to remove that assertion in PHPUnit. It's not worth adding an external package when we can just change a couple of lines of code.

Reply:
ok.

Reply:
Doing something like this should be enough for now:

```php
$this->assertArrayHasKey('database', $parameters);
$this->assertNull($parameters['database'], 'The "database" parameter is expected to be NULL for sentinels');

$this->assertArrayHasKey('password', $parameters);
$this->assertNull($parameters['password'], 'The "password" parameter is expected to be NULL for sentinels');
```